core/src/main/java/org/mvcspec/ozark/servlet/OzarkServletContextListener.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright © 2017 Ivar Grimstad (ivar.grimstad@gmail.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.mvcspec.ozark.servlet;
+
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.servlet.annotation.WebListener;
+
+import java.util.Set;
+import java.util.logging.Logger;
+
+import static org.mvcspec.ozark.servlet.OzarkContainerInitializer.CONTROLLER_CLASSES;
+
+/**
+ * Performs some sanity checks for controllers during startup
+ */
+@WebListener
+public class OzarkServletContextListener implements ServletContextListener {
+
+    private static final Logger log = Logger.getLogger(OzarkServletContextListener.class.getName());
+
+    private BeanManager beanManager;
+
+    @Inject
+    public OzarkServletContextListener(BeanManager beanManager) {
+        this.beanManager = beanManager;
+    }
+
+    @Override
+    public void contextInitialized(ServletContextEvent sce) {
+
+        ServletContext servletContext = sce.getServletContext();
+        Set<Class<?>> controllerClasses = (Set<Class<?>>) servletContext.getAttribute(CONTROLLER_CLASSES);
+
+        if (controllerClasses == null || controllerClasses.isEmpty()) {
+            return;
+        }
+
+        for (Class<?> controllerClass : controllerClasses) {
+            failIfNoCdiBean(controllerClass);
+            warnIfAnnotatedWithNamed(controllerClass);
+        }
+    }
+
+    /**
+     * Throws an exception if the passed controller class is not present in the CDI BeanManager.
+     * For example, this can happen with glassfish / Jersey if the application is using CDI discovery mode "annotated"
+     * and the controller class is not annotated with any scope annotation.
+     * Unlike in JAX-RS resources, MVC controllers are required to be CDI-managed beans.
+     */
+    private void failIfNoCdiBean(Class<?> controllerClass) {
+        Set<Bean<?>> controllerBeans = beanManager.getBeans(controllerClass);
+        if (controllerBeans == null || controllerBeans.isEmpty()) {
+            String message = String.format("The controller %s is not a managed CDI bean. Maybe the controller class is" +
```

Comment:
Minor issue: There is a space missing at the end of the first string literal. So the error message will end up like this:

> Maybe the controller class **ismissing** a scope annotation